compiler/rustc_mir_build/src/thir/constant.rs

```diff
@@ -20,58 +17,59 @@ pub(crate) fn lit_to_const<'tcx>(
         trace!("trunc {} with size {} and shift {}", n, width.bits(), 128 - width.bits());
         let result = width.truncate(n);
         trace!("trunc result: {}", result);
-        Ok(ConstValue::Scalar(Scalar::from_uint(result, width)))
+
+        Ok(ScalarInt::try_from_uint(result, width)
+            .unwrap_or_else(|| bug!("expected to create ScalarInt from uint {:?}", result)))
```

Comment:
why that change? seems unnecessary

Reply:
These conversion methods were missing on `ScalarInt`. Seemed reasonable to use this method here since we directly need a `ScalarInt`.

Reply:
ah, we go from `Scalar` to `ScalarInt` :sweat_smile:  sry